### PR TITLE
sync creature teleports

### PR DESCRIPTION
**Pull request by youbitchoc**
fixes vulture, deer, leviathan, mirosbird, mirosvulture

hook CreatureTakeFlight and RPC like we do SuckedIntoShortCut

### CODE REVIEW

File: Online/Entity/OnlineCreature.cs [hunk 1]
```diff
@@ -277,6 +277,51 @@ public void SuckedIntoShortCut(IntVector2 entrancePos, bool carriedByOther)
             enteringShortCut = false;
         }
 
+        public void BroadcastTookFlight(AbstractRoomNode.Type type, WorldCoordinate start, WorldCoordinate dest)
+        {
+            if (currentlyJoinedResource is RoomSession room)
+            {
+                RainMeadow.Debug(this);
+                if (id.type == 0) throw new InvalidProgrammerException("here");
```
Comment (6fears7) [inline Online/Entity/OnlineCreature.cs:285]: Intentional error message in the exception or residual from testing?

File: Game/RainMeadow.ShortcutHooks.cs [hunk 2]
```diff
@@ -157,6 +158,39 @@ private bool ShortcutHandlerOnVesselAllowedInRoom(On.ShortcutHandler.orig_Vessel
...
+                RainMeadow.Debug($"{onlineCreature} took flight");
+                onlineCreature.BroadcastTookFlight(type, start, dest);
+            }
+            else if (onlineCreature.enteringShortCut) // If this call was from a processing event
+            {
+                onlineCreature.enteringShortCut = false;
+            }
+            else
+            {
+                return;
```
Comment (6fears7) [inline Game/RainMeadow.ShortcutHooks.cs:189]: Maybe add a log here just so it's not a completely silent return

**Review by 6fears7 — COMMENTED** [review]
Tested this with miros birds yesterday. Wasn't able to force a scenario to test the desired conditions but it also didn't do anything super weird. If there's a spawn point with vultures for easy testing that would be ideal I think

File: Online/Entity/OnlineCreature.cs (hunk 1 above)
Comment (youbitchoc) [reply Online/Entity/OnlineCreature.cs:285]: copypasted from suckedintoshortcut

**Pull request merged**